examples/hikari_tanjun_basic_queue/music.py

```diff
@@ -0,0 +1,351 @@
+
+
+async def _skip_track(ctx: tanjun.abc.Context, lavalink: lavasnek_rs.Lavalink) -> None:
+    """Skips the current song."""
+    assert ctx.guild_id is not None
+
+    if not (skip := await lavalink.skip(ctx.guild_id)):
+        await ctx.respond("No tracks left to skip.")
+
+    elif node := await lavalink.get_guild_node(ctx.guild_id):
+        # If we skipped and the queue is empty we need to
+        # stop the player
+        if not node.queue and not node.now_playing:
+            await lavalink.stop(ctx.guild_id)
+
+        await ctx.respond(f"Skipped: {skip.track.info.title}")
+
+
+@music.with_slash_command
+@tanjun.as_slash_command("pause", "Pauses the current song.")
+async def pause_as_slash(
+    ctx: tanjun.abc.SlashContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    await _pause_playback(ctx, lavalink)
+
+
+@music.with_message_command
+@tanjun.as_message_command("pause")
+async def pause_as_message(
+    ctx: tanjun.abc.MessageContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    """Pauses the current song."""
+    await _pause_playback(ctx, lavalink)
+
+
+async def _pause_playback(ctx: tanjun.abc.Context, lavalink: lavasnek_rs.Lavalink) -> None:
+    """Pauses the current song."""
+    assert ctx.guild_id is not None
+
+    if node := await lavalink.get_guild_node(ctx.guild_id):
+        # Use node data to check if we are paused
+        if not (await node.get_data()).get("pause"):
+            # If we are playing, pause.
+            await node.set_data({"pause": True})
+            await lavalink.pause(ctx.guild_id)
+            await ctx.respond("Paused playback.")
+            return
+
+        else:
+            await ctx.respond("Playback is already paused.")
+            return
+
+    # We are not playing a track.
+    await ctx.respond("No song to pause, try playing one.")
+
+
+@music.with_slash_command
+@tanjun.as_slash_command("resume", "Resumes the current song.")
+async def resume_as_slash(
+    ctx: tanjun.abc.SlashContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    await _resume_playback(ctx, lavalink)
+
+
+@music.with_message_command
+@tanjun.as_message_command("resume")
+async def resume_as_message(
+    ctx: tanjun.abc.MessageContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    """Resumes the current song."""
+    await _resume_playback(ctx, lavalink)
+
+
+async def _resume_playback(ctx: tanjun.abc.Context, lavalink: lavasnek_rs.Lavalink) -> None:
+    """Resumes playing the current song."""
+    assert ctx.guild_id is not None
+
+    if node := await lavalink.get_guild_node(ctx.guild_id):
+        # Use node data to check if we are paused
+        if (await node.get_data()).get("pause"):
+            # We are paused, lets resume playback
+            await node.set_data({"pause": False})
```

Comment:
This has been fixed on lavalink-rs master, it shouldn't be needed in the next update to lavasnek.

Reply:
node.is_paused value never changes even when the player is paused. I have reverted `play`, `pause`, and `resume` to match the other very basic examples which just preform the action regardless of whether they are already paused to resumed etc.